### ISSUE TITLE
Improve password security for git2r package

### ISSUE THREAD

**Issue by jdblischak** [chirunconf/chirunconf19#15]
The [git2r](https://github.com/ropensci/git2r) package enables you to execute Git commands from R. I use this package a lot and have made small contributions in the past. I recently noticed that it is currently not possible to securely enter your password when pushing and pulling from a remote repository. I've opened an Issue (https://github.com/ropensci/git2r/issues/379) and would like to mentor a new contributor to enable this feature. If you're interested in working on this, please volunteer below.

**Comment by maurolepore** [comment]
Oh, given your experience with git2r I would like to pick your brain to see if you can help me find my credential on Windows. Right now my hack it to put this on my .Rprofile

```R
if (requireNamespace("git2r", quietly = TRUE)) {
  # https://happygitwithr.com/ssh-keys.html#ssh-keys
  cred <- git2r::cred_ssh_key(
    publickey =  "~/../.ssh/id_rsa.pub",
    privatekey = "~/../.ssh/id_rsa"
  )
}
```

**Comment by jdblischak** [comment]
@maurolepore I authenticate with HTTPS when I am working on Windows, but I'm happy to help troubleshoot. When was the last time you attempted to use SSH keys with git2r without your hack? It's my understanding that the SSH issue on Windows was due to the fact that R on Windows considers the home to be `~/Desktop`, so it can't find the keys in `~/.ssh`. I know some improvements to this have been merged (see discussion in https://github.com/ropensci/git2r/issues/353), so it may be worth trying again.

**Comment by jdblischak** [comment]
@anniew submitted the PR! https://github.com/ropensci/git2r/pull/383

**Comment by jdblischak** [comment]
This new feature is included in git2r version 0.25.1, which was recently released to CRAN a few days ago. Congrats again to @anniew!

https://github.com/ropensci/git2r/blob/master/NEWS#L34
